metricbeat/module/system/cpu/metrics/doc.go

```diff
@@ -0,0 +1,18 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metrics
```

Comment:
The package name `metrics` is maybe a little too generic for what is provided.

If you don't want to have this internal package to be used from the outside, put it into an internal package.

Reply:
I'm kind of on the fence about this. If we wanted to put it in `internal`, it should probably go somewhere else, maybe `metricbeat/system` ? Even if we consider these metrics to be producers purely for individual metricsets (or whatever comes after metricsets) There's going to be at least a little cross-polination between metrics, as occasionally something like `processes` will require memory data, and so on.

Reply:
If you want reuse, you need to move it somewhere else. If you want reuse at the system module level, consider to move the package to `metricbeat/module/system/internal/...`.

The critic is not only about the place, but `metrics` is quite generic, indicating that it can do more then collecting CPU and memory stats.

Reply:
Yah, that's a good point, the name `metrics` in general probably isn't good. I like `metricbeat/module/system/internal/`, so lets' go with that.